lib/galaxy/managers/session.py

```diff
@@ -6,16 +8,23 @@
     joinedload,
 )
 
+from galaxy.model.base import SharedModelMapping
+
+log = logging.getLogger(__name__)
+
 
 class GalaxySessionManager:
     """Manages GalaxySession."""
 
-    def __init__(self, model):
+    def __init__(self, model: SharedModelMapping):
         self.model = model
+        self.sa_session = model.context
```

Comment:
This gives you a single global session that is not going to be tied to a transaction.
```suggestion
    @property
    def sa_session(self):
        return self.model.session
```
This will give you the appropriate session.

Reply:
I will go through the codebase and replace all those occurences.

Reply:
Yeah - this is better. I don't know why I didn't think of this. Thanks!

Here we actually use ``self.model.GalaxySession`` because we cannot import GalaxySession directly because the code is reused by the tool shed with a different class. But in other instances it would be cool to inject a ``Callable[] -> Session`` to not even require a dependency on ``ModelMapping``.

Reply:
I agree, and I think with a custom identity check as in 8ab79eec43cb992565fb7bf6da4026ede8916ce6 we can do this.